lib/streamlit/delta_generator_singletons.py

```diff
@@ -0,0 +1,131 @@
+# Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022-2024)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+from contextvars import ContextVar, Token
+from typing import TYPE_CHECKING, Callable, Generic, TypeVar
+
+if TYPE_CHECKING:
+    from streamlit.delta_generator import DeltaGenerator
+    from streamlit.elements.lib.dialog import Dialog
+    from streamlit.elements.lib.mutable_status_container import StatusContainer
+
+"""
+The main purpose of this module (right now at least) is to avoid a dependency
+cycle between streamlit.delta_generator and some elements.
+"""
+
+
+main_dg: DeltaGenerator | None = None
+sidebar_dg: DeltaGenerator | None = None
+event_dg: DeltaGenerator | None = None
+bottom_dg: DeltaGenerator | None = None
```

Comment:
nit: should we prefix these with `_` since they should be used via the `get_` functions?

Reply:
I did not have it prefixed with `_` because it is imported in the `__init__.py` file, but I think you are right that it _usually_ should not be used. So I have added the prefix!